### PR TITLE
[DAT-950] - Add GetInvoice method

### PR DESCRIPTION
**Pull request by manu3210**
## Background

We need to create the GetInvoice method to get the invoice by customer and payment ID.

This code is part of the next flow:

```mermaid
sequenceDiagram
	participant MercadoPago
	participant MercadoPagoAPI
	participant BillingUserAPI
	participant BillingUserDB
	MercadoPago ->> BillingUserAPI: POST "/doppler-billing-user/Integration/{accountname}/MercadopagoNotification"
	BillingUserAPI->>MercadoPagoAPI: Get payment
    MercadoPagoAPI->>MercadoPago: Get payment
	MercadoPago->>MercadoPagoAPI: Payment info
	MercadoPagoAPI->>BillingUserAPI: Payment info
	BillingUserAPI->>BillingUserDB: Get invoice (BillingRepo)
	BillingUserDB->>BillingUserAPI: Invoice info
	BillingUserAPI->>BillingUserDB: Update invoice (BillingRepo)
	BillingUserAPI->>BillingUserDB: Get CreditCard (UserRepo)
	BillingUserDB->>BillingUserAPI: CC info
	BillingUserAPI->>BillingUserDB: Create payment (BillingRepo)
	BillingUserDB->>BillingUserAPI: Status info
	BillingUserAPI->>MercadoPago: OkStatusCode
```

Other pull requests that are part of the flow:

- [Create method UpdateInvoiceStatus](https://github.com/FromDoppler/doppler-billing-user/pull/174)
- [Add create payment method](https://github.com/FromDoppler/doppler-billing-user/pull/177)

## Changes

- Added GetInvoice method in BillingRepository

### CODE REVIEW

File: Doppler.BillingUser/Infrastructure/BillingRepository.cs [hunk 1]
```diff
@@ -808,6 +808,36 @@ [dbo].[BillingCredits] BC
             return billingCredit;
         }
 
+        public async Task<AccountingEntry> GetInvoice(int idClient)
```
Comment (andresmoschini) [inline Doppler.BillingUser/Infrastructure/BillingRepository.cs:811]: Creo que no es buena idea hacer PRs con código muerto (al menos por el momento).

Podemos seguir la conversación en otro PR donde hay un escenario similar: https://github.com/FromDoppler/doppler-billing-user/pull/177/files#r880381954

File: Doppler.BillingUser/Infrastructure/BillingRepository.cs [hunk 2]
```diff
@@ -808,6 +808,37 @@ [dbo].[BillingCredits] BC
...
+    AE.[IdAccountType],
+    AE.[IdInvoiceBillingType],
+    AE.[IdCurrencyType],
+    AE.[CurrencyRate],
+    AE.[Taxes]
+FROM
+    [dbo].[AccountingEntry] AE
+WHERE
+    idClient = @idClient AND AuthorizationNumber = @paymentId",
+        new
```
Comment (afantini) [inline Doppler.BillingUser/Infrastructure/BillingRepository.cs:834]: This doesn't seem to be well indented.

Comment (andresmoschini) [reply Doppler.BillingUser/Infrastructure/BillingRepository.cs:834]: @afantini you are talking about the last line, right?

```suggestion
                new
```

**Comment by jhoffman-ms** [comment]
👍

File: Doppler.BillingUser/Infrastructure/BillingRepository.cs (hunk 2 above)
Comment (afantini) [reply Doppler.BillingUser/Infrastructure/BillingRepository.cs]: Yes! sorry for the delay.

Comment (manu3210) [reply Doppler.BillingUser/Infrastructure/BillingRepository.cs]: No problem! you were right!. I modified it that before merging

**Pull request merged**